src/Entity/IdeasWorkshop/Idea.php

```diff
@@ -514,36 +542,41 @@ public function getDaysBeforeDeadline(): int
         }
 
         $date = clone $this->publishedAt;
-        $deadline = $date->add(new \DateInterval(self::PUBLISHED_INTERVAL));
+        $deadline = $date->add(new \DateInterval(self::FINALIZED_INTERVAL));
         $now = new Chronos();
 
         return $deadline <= $now ? 0 : $deadline->diff($now)->d;
     }
 
     public function isDraft(): bool
     {
-        return IdeaStatusEnum::DRAFT === $this->status;
+        return null === $this->publishedAt;
     }
 
     public function isPending(): bool
     {
-        return IdeaStatusEnum::PENDING === $this->status;
+        return new \DateTime('now') < $this->finalizedAt;
     }
 
     public function isFinalized(): bool
     {
-        return IdeaStatusEnum::FINALIZED === $this->status;
+        return new \DateTime('now') >= $this->finalizedAt;
     }
 
     public function isUnpublished(): bool
     {
-        return IdeaStatusEnum::UNPUBLISHED === $this->status;
+        return true !== $this->enabled;
     }
 
     public function publish(): void
     {
-        $this->status = IdeaStatusEnum::PENDING;
-        $this->publishedAt = new \DateTime();
+        $this->publishedAt = $now = new \DateTime();
+        $this->finalizedAt = $now->add(new \DateInterval(self::FINALIZED_INTERVAL));
```

Comment:
with this, you have the same dates in `publishedAt` and `finalizedAt` as in PHP the object passed by reference.
```php
$this->publishedAt = $now = new \DateTimeImmutable();

```